main.go

```diff
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/bitrise-io/go-utils/command"
+)
+
+const cacheArchivePath = "/tmp/cache-archive.tar"
+
+var (
+	gIsDebugMode = false
+)
+
+// StepParamsModel ...
+type StepParamsModel struct {
+	CacheAPIURL string
+	IsDebugMode bool
+}
+
+// GenerateDownloadURLRespModel ...
+type GenerateDownloadURLRespModel struct {
+	DownloadURL string `json:"download_url"`
+}
+
+// CreateStepParamsFromEnvs ...
+func CreateStepParamsFromEnvs() (StepParamsModel, error) {
+	stepParams := StepParamsModel{
+		CacheAPIURL: os.Getenv("cache_api_url"),
+		IsDebugMode: os.Getenv("is_debug_mode") == "true",
+	}
+
+	return stepParams, nil
+}
+
+func downloadCacheArchive(url string) error {
+	// Get the data
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			log.Printf(" [!] Failed to close Archive download response body: %s", err)
+		}
+	}()
+
+	if resp.StatusCode != 200 {
+		responseBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
```

Comment:
provide better error: `return fmt.Errorf("Failed to download archive - non success response code: %d, failed to read body, error: %s", error)`